reana_db/models.py

```diff
@@ -656,10 +656,28 @@ class Job(Base, Timestamp):
     env_vars = Column(JSONType)
     deleted = Column(Boolean)
     logs = Column(String, nullable=True)
+    started_at = Column(DateTime)
+    finished_at = Column(DateTime)
     prettified_cmd = Column(JSONType)
     job_name = Column(Text)
 
 
+@event.listens_for(Job.status, "set")
+def workflow_status_change_listener(job, new_status, old_status, initiator):
+    """Job status change listener."""
+    from .database import Session
+    if new_status != old_status:
+        if new_status in [
+            JobStatus.finished,
+            JobStatus.failed,
+        ]:
+            job.finished_at = datetime.now()
+        elif new_status in [JobStatus.running]:
+            job.started_at = datetime.now()
+
+    Session.commit()
```

Comment:
Should we commit in the _very rare_ case that `new_status` == `old_status`? Maybe this commit should only happen when they're different, what do you think?